### PR TITLE
Add validating webhook for ephemeral job

### PR DESCRIPTION
**Pull request by veophi**

### CODE REVIEW

File: pkg/webhook/ephemeraljob/validating/ephemeraljob_validating_handler.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+func validateEphemeralJobSpec(spec *appsv1alpha1.EphemeralJobSpec, path *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if spec.Selector != nil {
+		if spec.Selector.MatchLabels != nil || spec.Selector.MatchExpressions != nil {
+			if _, err := metav1.LabelSelectorAsSelector(spec.Selector); err != nil {
+				allErrs = append(allErrs, field.Invalid(path.Child("selector"), spec.Selector, err.Error()))
+			}
+		}
```
Comment (zmberg) [inline pkg/webhook/ephemeraljob/validating/ephemeraljob_validating_handler.go:67]: Does it not work if both are empty?

File: pkg/util/inplaceupdate/inplace_update_defaults.go [hunk 2]
```diff
@@ -265,16 +266,24 @@ func defaultCalculateInPlaceUpdateSpec(oldRevision, newRevision *apps.Controller
 			}
 			return nil
 		}
-		if op.Operation != "replace" || !containerImagePatchRexp.MatchString(op.Path) {
-			return nil
+		if op.Operation == "replace" && containerImagePatchRexp.MatchString(op.Path) {
```
Comment (zmberg) [inline pkg/util/inplaceupdate/inplace_update_defaults.go:269]: Why the modification? Does it have anything to do with validate?

File: pkg/util/inplaceupdate/inplace_update_defaults.go [hunk 3]
```diff
@@ -247,6 +247,7 @@
 		Revision:             newRevision.Name,
 		ContainerImages:      make(map[string]string),
 		ContainerRefMetadata: make(map[string]metav1.ObjectMeta),
+		ContainerResources:   make(map[string]v1.ResourceRequirements),
```
[the remark this answers is not in the document]
Comment (veophi) [reply pkg/util/inplaceupdate/inplace_update_defaults.go]: ignore

File: pkg/util/inplaceupdate/inplace_update_defaults.go (hunk 2 above)
Comment (veophi) [reply pkg/util/inplaceupdate/inplace_update_defaults.go]: ignore ...

File: pkg/webhook/ephemeraljob/validating/ephemeraljob_validating_handler.go (hunk 1 above)
Comment (veophi) [reply pkg/webhook/ephemeraljob/validating/ephemeraljob_validating_handler.go:67]: oh, yes, must not be empty.

**Pull request closed without merging**